core/services/relay/evm/read/batch.go

```diff
@@ -127,104 +127,183 @@ func newDefaultEvmBatchCaller(
+			results[idx].err = newErrorFromCall(
+				fmt.Errorf("%w: rpc call error: %s", types.ErrInternal, rpcBatchCalls[idx].Error),
+				call, block, true,
+			)
+
 			continue
 		}
 
-		if packedOutputs[i] == "" {
+		if hexEncodedOutputs[idx] == "" {
 			// Some RPCs instead of returning "0x" are returning an empty string.
 			// We are overriding this behaviour for consistent handling of this scenario.
-			packedOutputs[i] = "0x"
+			hexEncodedOutputs[idx] = "0x"
 		}
 
-		b, err := hexutil.Decode(packedOutputs[i])
+		packedBytes, err := hexutil.Decode(hexEncodedOutputs[idx])
 		if err != nil {
-			return nil, fmt.Errorf("decode result %s: packedOutputs %s: %w", call, packedOutputs[i], err)
+			callErr := newErrorFromCall(
+				fmt.Errorf("%w: hex decode result: %s", types.ErrInternal, err.Error()),
+				call, block, true,
+			)
+
+			callErr.Result = &hexEncodedOutputs[idx]
+
+			return nil, callErr
 		}
 
-		if err = c.codec.Decode(ctx, b, call.ReturnVal, codec.WrapItemType(call.ContractName, call.MethodName, false)); err != nil {
-			if len(b) == 0 {
-				results[i].err = fmt.Errorf("unpack result %s: %s: %w", call, err.Error(), errEmptyOutput)
+		if err = c.codec.Decode(
+			ctx,
+			packedBytes,
+			call.ReturnVal,
+			codec.WrapItemType(call.ContractName, call.MethodName, false),
+		); err != nil {
+			if len(packedBytes) == 0 {
+				callErr := newErrorFromCall(
+					fmt.Errorf("%w: %s: %s", types.ErrInternal, errEmptyOutput, err.Error()),
+					call, block, true,
+				)
+
+				callErr.Result = &hexEncodedOutputs[idx]
+
+				results[idx].err = callErr
 			} else {
-				results[i].err = fmt.Errorf("unpack result %s: %w", call, err)
+				callErr := newErrorFromCall(
+					fmt.Errorf("%w: codec decode result: %s", types.ErrInvalidType, err.Error()),
+					call, block, true,
+				)
+
+				callErr.Result = &hexEncodedOutputs[idx]
+				results[idx].err = callErr
 			}
+
 			continue
 		}
-		results[i].returnVal = call.ReturnVal
+
+		results[idx].returnVal = call.ReturnVal
 	}
 
 	return results, nil
 }
 
 func (c *defaultEvmBatchCaller) batchCallDynamicLimitRetries(ctx context.Context, blockNumber uint64, calls BatchCall) (BatchResult, error) {
 	lim := c.batchSizeLimit
+
 	// Limit the batch size to the number of calls
 	if uint(len(calls)) < lim {
 		lim = uint(len(calls))
 	}
+
 	for {
 		results, err := c.batchCallLimit(ctx, blockNumber, calls, lim)
 		if err == nil {
 			return results, nil
 		}
 
 		if lim <= 1 {
-			return nil, errors.Wrapf(err, "calls %+v", calls)
+			return nil, ErrRead{
+				Err:   fmt.Errorf("%w: limited call: call data: %+v", err, calls),
+				Batch: true,
+			}
 		}
 
 		newLim := lim / c.backOffMultiplier
 		if newLim == 0 || newLim == lim {
 			newLim = 1
 		}
+
 		lim = newLim
-		c.lggr.Errorf("retrying batch call with %d calls and %d limit that failed with error=%s",
-			len(calls), lim, err)
+
+		c.lggr.Debugf("retrying batch call with %d calls and %d limit that failed with error=%s", len(calls), lim, err)
```

Comment:
why change severity?

Reply:
If we have an error that cannot be handled, we return an error up the stack. Otherwise if we are handling the error, extra info is for debugging purposes. In this case we seem to be handling the error.